Client/Frontend/Browser/BrowserViewController.swift

```diff
@@ -1738,10 +1739,13 @@ extension BrowserViewController: TabManagerDelegate {
 
         if let tab = selected, let webView = tab.webView {
             updateURLBarDisplayURL(tab)
-            if tab.isPrivate != previous?.isPrivate {
-                applyTheme(tab.isPrivate ? .Private : .Normal)
+
+            if tab.type != previous?.type {
```

Comment:
What about just having the safety logic here inside of `applyTheme` then we can just "set" the theme whenever, and allow that to protect against overly resetting views?